app/views/teachers/LicenseStatsModal.coffee

```diff
@@ -0,0 +1,48 @@
+ModalComponent = require 'views/core/ModalComponent'
+LicenseStatsComponent = require('./components/LicenseStatsModal.vue').default
+
+module.exports = class LicenseStatsModal extends ModalComponent
+  id: 'license-stats-modal'
+  template: require('app/templates/core/modal-base-flat')
+  VueComponent: LicenseStatsComponent
+
+  constructor: (options) ->
+    super options
+    @prepaid = options.prepaid
+
+    @redeemers = @prepaid.get('redeemers') ? []
+    redeemerIds = @redeemers.map((r) =>
+      r.userID
+    )
+    @removedRedeemers = @prepaid.get('removedRedeemers') ? []
+    removedRedeemerIds = @removedRedeemers.map((r) =>
+      r.userID
+    )
+    @propsData =
+      hide: () => @hide()
+      loading: { finished: false }
+      redeemers: @redeemers
+      removedRedeemers: @removedRedeemers
+
+    @supermodel.resetProgress()
+    userNameRequest = @supermodel.addRequestResource 'user_names', {
```

Comment:
created hook in modal seems better way to do this. Any reason we are doing it here?